CMakeLists.txt

```diff
@@ -76,31 +135,42 @@ macro(set_lib list var value)
 	list(APPEND ${list} ${value})
 endmacro()
 
-# dylib - dymamic library (.so, .dll, .dylib)
-# staticlib - static library (.a, .lib)
-# implib - import library for windows dynamic library (DLL) - .lib
-# dylibs - list of files required for use dynamic libraty
-# staticlibs - list of files required for use static libraty
+# dylib[d] - dymamic library (.so, .dll, .dylib)
+# staticlib[d] - static library (.a, .lib)
+# implib[d] - import library for windows dynamic library (DLL) - .lib
+# dylibs[d] - list of files required for use dynamic libraty
+# staticlibs[d] - list of files required for use static libraty
 if(APPLE)
+	set_lib(dylibsd dylibd "libzenohcd.dylib")
```

Comment:
You can replace all that `If() ... elseif() ....` platform stuff for library file name generation using the following CMake variables:
https://cmake.org/cmake/help/v3.16/variable/CMAKE_SHARED_LIBRARY_SUFFIX.html
https://cmake.org/cmake/help/v3.16/variable/CMAKE_STATIC_LIBRARY_SUFFIX.html
https://cmake.org/cmake/help/v3.16/variable/CMAKE_SHARED_LIBRARY_PREFIX.html
https://cmake.org/cmake/help/v3.16/variable/CMAKE_STATIC_LIBRARY_PREFIX.html

Reply:
Done